### DISCUSSION TITLE
How to control unit for timestamp columns in parquet file target?

### DISCUSSION THREAD

**Discussion by nixent** [slingdata-io/sling-cli#184] [Q&A] [answered]
When target is parquet file Sling writes timestamp columns with ns precision timestamp("ns"). I'd like to change that to timestamp("ms"). How to do that?

**Comment by flarco** [comment]
Not yet implemented. Will do for next release.

Curious, how can `ns` be disadvantageous? My thought was it would cover all the precision you need, so make it by default.

**Reply by nixent** [reply]
I'm converting parquet files created by Sling to delta table format. Current implementation of delta-rs doesn't support nanosecond precision and fails. Now I'm trying to compensate until [1721](https://github.com/delta-io/delta-rs/issues/1721) is fixed. I can handle it in my code as well, no big issue. Thank you

**Reply by flarco** [reply, marked as answer]
k, next release should support it by defining the column type in `source.options.columns`:
- `timestamp(3)`
- `timestamp(6)`
- `timestamp(9)`. 

`timestamp` will default to `timestamp(9)` for parquet.
